src/Lender.sol

```diff
@@ -548,20 +519,20 @@ contract Lender {
         // Fetch the principal token for this lend call
         address pt = IMarketPlace(marketPlace).markets(u, m, p);
 
-        // Get the starting balance for the Lender
-        uint256 starting = IERC20(pt).balanceOf(address(this));
-
         // Conduct the lend operation to acquire principal tokens
-        (bool success, ) = adapter.delegatecall(
+        (bool success, bytes memory returndata) = adapter.delegatecall(
```

Comment:
This is just a first thought -- the calldata tells the adapter how much to spend right?

But what if the user creates calldata that tells it to spend more than the `a` amount?

I feel like we may need to pass the a amount to the adapter as well? Both for this verification (perhaps) and for calculation / removal of the fee?

Reply:
This is a good catch. In fact, the parameter amount parameter that was introduced is unnecessary. Instead, it should be supplied via the `d` parameter and the adapter to validate that sufficient funds are transferred by the user to execute the order. This is now fixed.